python/src/com/jetbrains/python/codeInsight/imports/PythonImportUtils.java

```diff
@@ -234,20 +237,31 @@ private static boolean isPossibleModuleReference(PyElement node) {
 
   private static Collection<PsiElement> findImportableModules(PsiFile targetFile, String reftext, Project project, GlobalSearchScope scope) {
     List<PsiElement> result = new ArrayList<>();
-    PsiFile[] files = FilenameIndex.getFilesByName(project, reftext + ".py", scope);
-    for (PsiFile file : files) {
-      if (isImportableModule(targetFile, file)) {
-        result.add(file);
+    //add packages
+    final PsiManager psiMgr = PsiManager.getInstance(project);
+    FileBasedIndex.getInstance().processValues(FilenameIndex.NAME, reftext, null, (file, value) -> {
```

Comment:
In the use case I'm personally interested in it doesn't make a difference. So for me it would be ok to use 'processFilesByName'.  However,   I've also tested it with mixed Python and non-Python projects with the Python plugin. In the cases where most directories are not Python packages it does make a considerable difference.